kuma/spam/forms.py

```diff
@@ -0,0 +1,67 @@
+from django import forms
+from tower import ugettext_lazy as _
+import waffle
+
+from . import akismet
+
+SPAM_CHECKS_SWITCH = 'spam_checks_enabled'
+
+
+class AkismetFormMixin(object):
+    """
+    The main form mixin for Akismet checks.
+
+    Form classes using this can reimplement the methods starting with
+    "akismet_" below to extend its functionality.
+    """
+    akismet_client = akismet.Akismet()
+    akismet_error_message = _('The submitted data contains invalid content.')
+
+    def __init__(self, request, *args, **kwargs):
+        self.request = request
+        super(AkismetFormMixin, self).__init__(*args, **kwargs)
+
+    def akismet_parameters(self):
+        """
+        When using this mixin make sure you implement this method,
+        get the parent class' value and return a dictionary of
+        parameters matching the ones of the Akismet.check_comment method.
+
+        Use the self.instance or self.request variables to build it.
+        """
+        if not hasattr(self, 'cleaned_data'):
+            raise forms.ValidationError(
+                _('The form data has not yet been validated. '
+                  'Please try again.')
+            )
+        return {
+            'user_ip': self.request.META.get('REMOTE_ADDR', ''),
+            'user_agent': self.request.META.get('HTTP_USER_AGENT', ''),
+            'referrer': self.request.META.get('HTTP_REFERER', ''),
+        }
+
+    def akismet_enabled(self):
+        """
+        Decides whether to even check for spam during the form validation.
+
+        Checks the API client if it's ready by default.
+        """
+        return (self.akismet_client.ready and
+                waffle.switch_is_active(SPAM_CHECKS_SWITCH))
+
```

Comment:
The order should be reversed - check if the switch is active, then make a network request to the Akismet service